internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -374,6 +374,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 
 		cm.SetConditions(xpv1.Deleting())
 		if meta.WasCreated(cp) {
+			if meta.WasDeleted(cp) {
+				if *cm.GetCompositeDeletePolicy() == xpv1.CompositeDeleteForeground {
```

Comment:
Would it make sense to check for the `foregroundDeletion` finalizer having been set on the composite here instead? I imagine what we're trying to detect is that the composite is currently undergoing foreground deletion, and the presence of that finalizer seems like a more 'direct' signal.

This is late in the game so I'm happy to ignore the thought (or follow up in a future PR).

Reply:
That would probably work but I would want to test it thoroughly.  I'm happy to put a PR in for 1.11 and test it before it releases.